### PR TITLE
[Telemetry] KubeRay version and CRD

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

Screen Shot 2024-03-16 at 12 19 23 AM

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/controllers/ray/common/pod.go [hunk 1]
```diff
@@ -629,6 +629,13 @@ func setContainerEnvVars(pod *corev1.Pod, rayNodeType rayv1.RayNodeType, rayStar
 		usageEnv := corev1.EnvVar{Name: utils.RAY_USAGE_STATS_KUBERAY_IN_USE, Value: "1"}
 		container.Env = append(container.Env, usageEnv)
 	}
+	if rayNodeType == rayv1.HeadNode {
+		extraTagsEnv := corev1.EnvVar{
+			Name:  utils.RAY_USAGE_STATS_EXTRA_TAGS,
```
Comment (kevin85421) [inline ray-operator/controllers/ray/common/pod.go:634]: Do we need to inject this env var to worker nodes?

File: ray-operator/main.go [hunk 2]
```diff
@@ -37,11 +37,8 @@ import (
 )
 
 var (
-	_version_   = "0.2"
```
Comment (kevin85421) [inline ray-operator/main.go:40]: The changes in this file are unrelated to the telemetry. I simply deleted some unused code.

File: ray-operator/controllers/ray/common/pod.go (hunk 1 above)
Comment (jjyao) [reply ray-operator/controllers/ray/common/pod.go:634]: No, head node is enough.

File: ray-operator/controllers/ray/common/pod.go [hunk 3]
```diff
@@ -629,6 +629,13 @@ func setContainerEnvVars(pod *corev1.Pod, rayNodeType rayv1.RayNodeType, rayStar
...
+			Value: fmt.Sprintf("kuberay_version=%s,crd=%s", utils.KUBERAY_VERSION, string(creatorCRDType)),
```
Comment (jjyao) [inline ray-operator/controllers/ray/common/pod.go:635]: Let's make the key kuberay_version and kuberay_crd

**Review by jjyao — APPROVED** [review]
Let's test it manually

File: ray-operator/controllers/ray/utils/constant.go [hunk 4]
```diff
@@ -167,6 +168,9 @@ const (
 
 	// RayNodeHeadGroupLabelValue is the value for the RayNodeGroupLabelKey label on a head node
 	RayNodeHeadGroupLabelValue = "headgroup"
+
+	// Telemetry
+	KUBERAY_VERSION = "nightly"
```
Comment (andrewsykim) [inline ray-operator/controllers/ray/utils/constant.go:173]: @kevin85421 since this is defined as a constant, it will always be `nightly` instead of the actual kuberay version. Is that expected or am I missing something?

Comment (kevin85421) [reply ray-operator/controllers/ray/utils/constant.go:173]: I opened a PR to the release branch to update it #2036.
